src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts

```diff
@@ -108,9 +119,14 @@ const $signatoryAccounts = combine(
     const allSignatories = walletUtils.getAccountsBy(wallets, ({ accountId }) => signatoriesMap[accountId]);
     const uniqueSignatories = uniqBy(allSignatories, 'accountId');
 
-    return wallet.accounts[0].signatories.filter((signatory) => {
-      return uniqueSignatories.some((s) => s.accountId === signatory.accountId);
-    });
+    return wallet.accounts[0].signatories.reduce<Signatory[]>((acc, signatory) => {
+      const matchingSignatory = uniqueSignatories.find((s) => s.accountId === signatory.accountId);
```

Comment:
Can use `dictionary` for `signatoriesMap` and replace `reduce` with `filter > map`